exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,171 @@
+
+        if (inPenaltyBox && !rollIsEven) {
+            isGettingOutOfPenaltyBox = true;
+            System.out.println(players.get(currentPlayer) + " is getting out of the penalty box");
+            updatePlayerLocationAndCategory(roll);
+            askQuestion();
+        } else if (inPenaltyBox) {
+            System.out.println(players.get(currentPlayer) + " is not getting out of the penalty box");
+            isGettingOutOfPenaltyBox = false;
+        } else {
+            updatePlayerLocationAndCategory(roll);
+            askQuestion();
+        }
+    }
+
+    private void updatePlayerLocationAndCategory(int roll) {
+        places[currentPlayer] = places[currentPlayer] + roll;
+
+        if (places[currentPlayer] > 11)
+            places[currentPlayer] = places[currentPlayer] - 12;
+
+        System.out.println(players.get(currentPlayer)
+                + "'s new location is "
+                + places[currentPlayer]);
+        System.out.println("The category is " + currentCategory());
+    }
+
+    private void askQuestion() {
+        List<String> currentQuestions = listOfCategories.getMapOfListsOfQuestions().get(currentCategory());
+        System.out.println(currentQuestions.remove(0));
+    }
+
+
+    private String currentCategory() {
+        int playerPlace = places[currentPlayer];
+
+        if (playerPlace > 7) {
+            playerPlace = playerPlace - 8;
+        } else if (playerPlace > 3) {
+            playerPlace = playerPlace - 4;
+        }
+
+        switch (playerPlace) {
+            case 0:
+                return POP;
+            case 1:
+                return SCIENCE;
+            case 2:
+                return SPORTS;
+            default:
+                return ROCK;
+        }
+    }
+
+    public boolean wasCorrectlyAnswered() {
+        boolean inPenaltyBoxAfterQuestion = this.inPenaltyBox[currentPlayer];
+
+        if (inPenaltyBoxAfterQuestion && isGettingOutOfPenaltyBox) {
+            updateCoins("Answer was correct!!!!");
+
+            boolean winner = didPlayerWin();
+
+            updatePlayersPlace();
+            return winner;
+
+        } else if (inPenaltyBoxAfterQuestion) {
+            updatePlayersPlace();
+            return true;
+        } else {
+            updateCoins("Answer was corrent!!!!");
+
+            boolean winner = didPlayerWin();
+            updatePlayersPlace();
+
+            return winner;
+        }
+    }
+
+    private void updatePlayersPlace() {
+        currentPlayer++;
+        if (currentPlayer == players.size()) {
+            currentPlayer = 0;
+        }
+    }
+
+    private void updateCoins(String s) {
+        System.out.println(s);
+        purses[currentPlayer]++;
+        System.out.println(players.get(currentPlayer)
+                + " now has "
+                + purses[currentPlayer]
+                + " Gold Coins.");
+    }
+
+    public boolean isAnswerWrong() {
+        System.out.println("ListOfCategories was incorrectly answered");
```

Comment:
The first word of this string may have been changed to 'ListOfCategories' during a refactor, this might need a quick look.

Reply:
done 👍